chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/impl/DeveloperManagerImpl.java

```diff
@@ -775,4 +779,85 @@ public String toString() {
                     origChplProductNumberB);
         }
     }
+
+    @Override
+    public void validateDeveloperInSystemIfExists(final PendingCertifiedProductDetails pendingCp)
+            throws EntityRetrievalException, ValidationException {
+        if (!isNewDeveloperCode(pendingCp.getChplProductNumber())) {
+            DeveloperDTO systemDeveloperDTO = null;
+            if (pendingCp.getDeveloper() != null && pendingCp.getDeveloper().getDeveloperId() != null) {
+                systemDeveloperDTO = getById(pendingCp.getDeveloper().getDeveloperId());
+            }
+            if (systemDeveloperDTO != null) {
+                final Object pendingAcbNameObj = pendingCp.getCertifyingBody()
+                        .get(CertifiedProductSearchDetails.ACB_NAME_KEY);
+                if (pendingAcbNameObj != null && !StringUtils.isEmpty(pendingAcbNameObj.toString())) {
+                    Set<String> sysDevErrorMessages = runSystemValidations(systemDeveloperDTO,
+                            pendingAcbNameObj.toString());
+                    if (!sysDevErrorMessages.isEmpty()) {
+                        throw new ValidationException(sysDevErrorMessages);
+                    }
+                } else {
+                    LOGGER.error("Unable to validate system developer as the pending ACB Name is null "
+                            + "or its String representation is null or empty");
+                    throw new ValidationException(msgUtil.getMessage("system.developer.pendingACBNameNullOrEmpty"));
+                }
+            } else {
+                LOGGER.warn("Skipping system validation due to null pending developer or a null system developer");
+            }
+        } else {
+            LOGGER.info("Skipping system validation due to new developer code '" + NEW_DEVELOPER_CODE + "'");
+        }
+    }
+
+    private boolean isNewDeveloperCode(final String chplProductNumber) {
+        String devCode = chplProductNumberUtil.getDeveloperCode(chplProductNumber);
+        return !StringUtils.isEmpty(devCode) && devCode.contentEquals(NEW_DEVELOPER_CODE);
```

Comment:
I think we usually use something like `StringUtils.equals(devCode, NEW_DEVELOPER_CODE)`, which handles the case where devCode might be null, in places like this